src/Components/CreateImageWizard/steps/Users/component/UserInfo.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { useState } from 'react';
+
+import {
+  Button,
+  Checkbox,
+  Form,
+  FormGroup,
+  FormHelperText,
+  HelperText,
+  HelperTextItem,
+} from '@patternfly/react-core';
+import { ExternalLinkAltIcon } from '@patternfly/react-icons';
+import { ActionCreatorWithPayload } from '@reduxjs/toolkit';
+
+import { useAppDispatch, useAppSelector } from '../../../../../store/hooks';
+import {
+  selectUsers,
+  UserWithAdditionalInfo,
+  editUser,
+  addUser,
+  selectUserName,
+  selectUserPassword,
+  selectConfirmUserPassword,
+  selectUserSshKey,
+  selectUserAdministrator,
+  setUserName,
+  setUserPassword,
+  setConfirmUserPassword,
+  setUserSshKey,
+  changeUserAdministrator,
+  selectUserindex,
+} from '../../../../../store/wizardSlice';
+import { useUserValidation } from '../../../utilities/useValidation';
+import {
+  HookValidatedInput,
+  HookValidatedInputWithButton,
+  HookValidatedTextArea,
+} from '../../../ValidatedTextInput';
+const UserInfo = () => {
+  const dispatch = useAppDispatch();
+  const users = useAppSelector(selectUsers);
+  const userName = useAppSelector(selectUserName);
+  const index = useAppSelector(selectUserindex);
+  const userPassword = useAppSelector(selectUserPassword);
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+  const [isConfirmPasswordVisible, setIsConfirmPasswordVisible] =
+    useState(false);
+  const confirmUserPassword = useAppSelector(selectConfirmUserPassword);
+  const userSshKey = useAppSelector(selectUserSshKey);
+  const userAdministrator = useAppSelector(selectUserAdministrator);
+  const handleFieldChange = <K extends keyof UserWithAdditionalInfo>(
+    field: K,
+    value: UserWithAdditionalInfo[K],
+    setterAction?: ActionCreatorWithPayload<
+      string | boolean | undefined | number
+    >,
+    additionalLogic?: (updatedUser: UserWithAdditionalInfo) => void
+  ) => {
+    // Create updated user object
+    const updatedUser: UserWithAdditionalInfo = {
+      index,
+      name: userName,
+      password: userPassword,
+      confirmPassword: confirmUserPassword,
+      ssh_key: userSshKey,
+      administrator: userAdministrator,
+      [field]: value,
+    };
+    // Call specific setter action if provided
+    if (setterAction) {
+      dispatch(setterAction(value));
+    }
+    // Optional additional logic
+    if (additionalLogic) {
+      additionalLogic(updatedUser);
+    }
+    // Determine if user exists by index, not name
```

Comment:
Why not name? Shouldn't the user name be unique? :thinking:

Reply:
Ah right, yeah, that makes sense :+1:

Reply:
I remove this function